compiler/x/codegen/OMRTreeEvaluator.cpp

```diff
@@ -5962,6 +5963,162 @@ OMR::X86::TreeEvaluator::bitpermuteEvaluator(TR::Node *node, TR::CodeGenerator *
    return resultReg;
    }
 
+static TR::Register*
+compressExpandBitsEvaluator(TR::Node *node, TR::CodeGenerator *cg, TR::InstOpCode::Mnemonic regRegRegOpCode, TR::InstOpCode::Mnemonic regRegMemOpCode)
+   {
+   TR::Node *srcNode = node->getFirstChild();
+   TR::Node *maskNode = node->getSecondChild();
+   TR::Register *srcReg = cg->longClobberEvaluate(srcNode);
+
+   if (maskNode->getReferenceCount() == 1 &&
+       maskNode->getRegister() == NULL &&
+       maskNode->getOpCode().isLoadVar())
+      {
+      TR::MemoryReference *maskMR = generateX86MemoryReference(maskNode, cg);
+      generateRegRegMemInstruction(regRegMemOpCode, node, srcReg, srcReg, maskMR, cg);
+
+      maskMR->decNodeReferenceCounts(cg);
+      }
+   else
+      {
+      TR::Register *maskReg = cg->evaluate(maskNode);
+      generateRegRegRegInstruction(regRegRegOpCode, node, srcReg, srcReg, maskReg, cg);
+
+      cg->stopUsingRegister(maskReg);
+      cg->decReferenceCount(maskNode);
+      }
+
+   node->setRegister(srcReg);
+   cg->decReferenceCount(srcNode);
+
+   return srcReg;
+   }
+
+static TR::Register*
+smallCompressBitsEvaluator(TR::Node *node, TR::CodeGenerator *cg, const bool isShort)
+   {
+   // bcompressbits and scompressbits require special logic to handle zero-extension
+   TR_ASSERT_FATAL(node->getDataType() == TR::Int16 || node->getDataType() == TR::Int8, "smallCompressBitsEvaluator only supports 16, or 8 bits");
+
+   TR::Node *srcNode = node->getFirstChild();
+   TR::Node *maskNode = node->getSecondChild();
+   TR::Register *srcReg = NULL;
+   TR::Register *maskReg = NULL;
+   TR::MemoryReference *maskMR = NULL;
+   TR::Register *resReg = NULL;
+
+   // Only one of src or mask needs to be zero-extended
+   // First check if either can be zero-extended at compile time
+   if (maskNode->getOpCode().isLoadConst())
+      {
+      uint32_t mask = isShort ? maskNode->getUnsignedShortInt() : maskNode->getUnsignedByte();
+      maskReg = TR::TreeEvaluator::loadConstant(maskNode, mask, TR_RematerializableInt, cg);
+      resReg = maskReg;
+      }
+   else if (srcNode->getOpCode().isLoadConst())
+      {
+      uint32_t src = isShort ? srcNode->getUnsignedShortInt() : srcNode->getUnsignedByte();
+      srcReg = TR::TreeEvaluator::loadConstant(srcNode, src, TR_RematerializableInt, cg);
+      resReg = srcReg;
+      }
+   // Otherwise check if src is in memory and can be therefore be loaded with a zero-extending opcode
+   else if (srcNode->getReferenceCount() == 1 && srcNode->getRegister() == NULL && srcNode->getOpCode().isLoadVar())
+      {
+      TR::InstOpCode::Mnemonic extendOpCode = isShort ? TR::InstOpCode::MOVZXReg4Mem2 : TR::InstOpCode::MOVZXReg4Mem1;
+      TR::MemoryReference *srcMR = generateX86MemoryReference(maskNode, cg);
```

Comment:
Should this be `srcNode` rather than `maskNode` ?

Apologies, but I think I asked this before and if you answered it I don't recall your response.

Reply:
Yes that should be `srcNode`. I did a whole refactor and made the same mistake again 🤦‍♂️. I'll get that fixed.